### PR TITLE
refactor aws nodeAddresses tests

### PR DESCRIPTION
**Pull request by gongguan**
Ref https://github.com/kubernetes/kubernetes/pull/90894#discussion_r424561146

/kind cleanup

```release-note
None
```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws_test.go [hunk 1]
```diff
@@ -583,68 +583,43 @@ func testHasNodeAddress(t *testing.T, addrs []v1.NodeAddress, addressType v1.Nod
...
-			PrivateIpAddresses: []*ec2.InstancePrivateIpAddress{
-				{
-					PrivateIpAddress: aws.String("192.168.0.1"),
+	return ec2.Instance{
+		InstanceId:       aws.String(fmt.Sprintf("i-%d", num)),
+		PrivateDnsName:   aws.String(privateDNSName),
+		PrivateIpAddress: aws.String(privateIP),
+		PublicDnsName:    aws.String(publicDNSName),
+		PublicIpAddress:  aws.String(publicIP),
+		NetworkInterfaces: []*ec2.InstanceNetworkInterface{
```
Comment (wojtek-t) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws_test.go:598]: It seems that NetworkInterfaces was set only for the instance0 originally.
Which for NodeAddresses - actually should make some difference?

Comment (gongguan) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws_test.go:598]: I had thought it's omitted in others. Should it filled with a mark(only instance0 set) or filled if privateIP not null?

Comment (gongguan) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws_test.go]: > It seems that NetworkInterfaces was set only for the instance0 originally.
> Which for NodeAddresses - actually should make some difference?

Set a new mark `setNetInterface` which makes instance0 different.

**Comment by wojtek-t** [comment]
Noop change - thanks!

/lgtm
/approve

**Pull request merged**